internal/server/middlewares.go

```diff
@@ -118,17 +132,16 @@ func requestLogger(lg *zap.Logger) middleware {
 			reader := io.NopCloser(bytes.NewBuffer(buf))
 			req.Body = reader
 
-			body := json.RawMessage(buf)
-			jsonBody, err := json.Marshal(body)
-			if err != nil {
-				lg.Debug("error marshling request body: %v", zap.String("error", err.Error()))
-				http.Error(wr, err.Error(), http.StatusInternalServerError)
-				return
-			}
-
-			clientID, _, _ := req.BasicAuth()
+			if len(buf) > 0 {
+				dst := &bytes.Buffer{}
+				if err := json.Compact(dst, buf); err != nil {
+					lg.Debug("error json compacting request body: %v", zap.String("error", err.Error()))
+					http.Error(wr, err.Error(), http.StatusInternalServerError)
```

Comment:
let's not return golang specific error to client, especially during internal server error.
Also always make sure the error response from our API are always in the same format.

use this instead
```
utils.WriteErrMsg(w, http.StatusInternalServerError, "Internal Server Error")
```

Reply:
don't forget to also update another error handling in the same function.

Reply:
correct, thanks!

Reply:
even we don't really need to error out in this case, just log the error, since this is only a logging middleware.

Reply:
+1 @mabdh on that, I am also not sure we need to return error here, we can just log the error if it happens